gocourse02/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"fmt"
+)
+
+func main() {
+	zookeeper := NewZookeeper(false, "Mukola")
+	zookeeper.PrintName()
+
+	cage := NewCage(true, 1)
+	fmt.Printf("Cage number %d state =>%t\n", cage.Number, cage.IsBroken)
+	if !zookeeper.IsBusy && cage.IsBroken {
+		zookeeper.FixCage(cage)
+	}
+	fmt.Printf("Cage number %d state =>%t\n", cage.Number, cage.IsBroken)
+
+	var sirko = NewAnimal("Sirko", cage, Cat)
+	fmt.Printf("Animal with name %s is sitting in cage number %d type of %s\n", sirko.Name, sirko.Cage.Number, sirko.AnimalType)
+
+	var sirkoSon = NewAnimal("Sirko junior", cage, Tiger)
+	fmt.Printf("Animal with name %s is sitting in cage number %d type of %s\n", sirkoSon.Name, sirkoSon.Cage.Number, sirkoSon.AnimalType)
+
+	var sirkoDaughter = NewAnimal("", nil, Manul)
+	if sirkoDaughter == nil {
+		fmt.Println("Everything work as expected")
+	}
+
+}
+
+const Cat = "Cat"
+const Manul = "Manul"
+const Tiger = "Tiger"
```

Comment:
Ось так має бути краще:
```suggestion
type AnimalType string

const Cat AnimalType = "Cat"
const Manul AnimalType = "Manul"
const Tiger AnimalType = "Tiger"
```

Про константи почитати тут https://go.dev/doc/effective_go#constants і тут https://go.dev/blog/constants